time-machine/main.py

```diff
@@ -0,0 +1,138 @@
+
+
+def setup_logger() -> logging.Logger:
+    urllib3_logger = logging.getLogger("urllib3")
+    urllib3_logger.setLevel(logging.WARNING)
+
+    logger = logging.getLogger()
+    logger.setLevel(logging.INFO)
+    formatter = logging.Formatter('%(asctime)s [%(levelname)s] %(name)s: %(message)s')
+
+    file_handler = logging.FileHandler('app.log', mode='a')
+    file_handler.setFormatter(formatter)
+    logger.addHandler(file_handler)
+
+    console_handler = logging.StreamHandler()
+    console_handler.setLevel(logging.INFO)
+    console_handler.setFormatter(formatter)
+    logger.addHandler(console_handler)
+
+    return logger
+
+
+logger = setup_logger()
+
+
+def load_api_configuration() -> Configuration:
+    logger.info("Loading API configuration.")  # Reduced message
+    config = Configuration(
+        host=os.getenv('VISIER_HOST'),
+        api_key=os.getenv('VISIER_APIKEY'),
+        username=os.getenv('VISIER_USERNAME'),
+        password=os.getenv('VISIER_PASSWORD'),
+        client_id=os.getenv('VISIER_CLIENT_ID'),
+        client_secret=os.getenv('VISIER_CLIENT_SECRET'),
+        redirect_uri=os.getenv('VISIER_REDIRECT_URI'),
+        vanity=os.getenv('VISIER_VANITY'))
+    return config
+
+
+def parse_args():
+    logger.info(f"Parsing command line arguments: {sys.argv[1:]}.")  # Add arguments to log
+    parser = argparse.ArgumentParser(description="Time Machine sample.")
+    parser.add_argument('-q', '--query_file_path', type=str, required=True,
+                        help="File path to aggregate query definition.")
+    parsed_args = parser.parse_args()
+    return parsed_args
+
+
+def get_temp_file_path(args):
+    file_name, _ = os.path.splitext(os.path.basename(args.query_file_path))
+    storage_path = os.getenv('STORAGE_PATH', default='tmp_storage')
+    date_timestamp = datetime.now().strftime('%Y%m%d_%H%M%S')
+    temp_data_file = os.path.join(storage_path, f'{file_name}_{date_timestamp}.csv')
+    return temp_data_file
+
+
+def download_data(config: Configuration,
+                  query_file_path: str,
+                  data_file_path: str):
+    """
+    Downloads metrics data and saves it to a file.
+    Args:
+        config: The API configuration object.
+        query_file_path: The file path to the aggregate query definition.
+        data_file_path: The file path to save the downloaded data.
+    """
+    with open(query_file_path) as f:
+        headcount_json = f.read()
+    query_dto = AggregationQueryExecutionDTO.from_json(headcount_json)
+
+    api_client = ApiClient(config, 'Accept', 'text/csv')
+    query_api = DataQueryApi(api_client)
+
+    logger.info(f"Executing query: {query_file_path}")  # Combined messages
+    response = query_api.aggregate_without_preload_content(query_dto)
+    if not 200 <= response.status <= 299:
+        raise Exception(f"API request failed with status code {response.status}: {response.data.decode()}")
+
+    os.makedirs(os.path.dirname(data_file_path), exist_ok=True)
```

Comment:
Consider tracking whether you're actually creating a new directory and if so, delete the dir too before existing

Reply:
Added removing dir